dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -81,9 +82,9 @@
     return inputClassPath;
   }
 
-  private static SyntheticRepository createSyntheticRepository(List<Path> paths) {
+  private static Repository createClassRepository(List<Path> paths) {
     ClassPath classPath = new LinkageCheckClassPath(paths);
-    return SyntheticRepository.getInstance(classPath);
+    return new ClassPathRepository(classPath);
```

Comment:
What's the difference between ClassPathRepository and SyntheticRepository?

Reply:
Looks like it doesn't clear classes out of the cache as aggressively?

Reply:
Right. Culprit was BCEL's MemorySensitiveClassPathRepository, which maintains map of soft references (className -> JavaClass). The entries are unnecessarily targeted by GC.

When I changed it to BCEL's ClassPathRepository, which does not use soft references, the test became faster (it finishes in 30 secs).